spec.md

```diff
@@ -3679,7 +3706,7 @@ statement =
   | "regreset" , id , ":" , type , "," , expr , "," , expr , "," , expr ,
     [info]
   | memory
-  | "inst" , id , "of" , id , [ info ]
+  | ( "inst" | "remote_inst" ) , id , "of" , id , [ info ]
```

Comment:
would this be better expressed as a more general concept of "usage group" of modules? where passes might decide to do things to different groups based on their type of instantiation (omit them entirely, move them to different directories, do them as binds, keep them present, do inlining...)

Reply:
I'm trying to leave the door open for this without solving it right here. This is what the following language in ABI doc is hinting at:

> Future changes to this ABI document will enable more fine-grained control of remote instantiation.

I was talking with @darthscsi about this earlier and he was proposing:

```
group verif : 
  inst bar of Bar
```

As I didn't want to start allowing arbitrary things inside the group other than instances, he preferred remote instantiation over an instance-only group.

That said, we need some answer for how to put things in different bind files and how grouping works.

Reply:
I guess to me the moth logical way to think would be 

inst foo of Foo group "amazing"


Then something in compiler cmd line /ABI could decide in this lowering that "amazing" should be a verilog-style remote instantiation, or omitted entirely, or left in place.

Just seems strange to put such a verilog-specific lowering so tightly into the IR, but still with insufficient information to make use of it (if we agree the idea of grouping is useful)

Reply:
I'm playing with the `group` angle which may be a better way to do this.

I agree with what you're saying. This PR is currently providing a mechanism for remote instantiation without a mechanism for conditional instantiation. We likely need the latter.